core/chains/evm/client/rpc_client.go

```diff
@@ -180,26 +166,9 @@ func (r *RPCClient) Ping(ctx context.Context) error {
 	return err
 }
 
-func (r *RPCClient) UnsubscribeAllExcept(subs ...multinode.Subscription) {
-	r.subsSliceMu.Lock()
-	defer r.subsSliceMu.Unlock()
-
-	keepSubs := map[multinode.Subscription]struct{}{}
-	for _, sub := range subs {
-		keepSubs[sub] = struct{}{}
-	}
-
-	for sub := range r.subs {
-		if _, keep := keepSubs[sub]; !keep {
-			sub.Unsubscribe()
-			delete(r.subs, sub)
-		}
-	}
-}
-
 // Not thread-safe, pure dial.
 func (r *RPCClient) Dial(callerCtx context.Context) error {
-	ctx, cancel := r.makeQueryCtx(callerCtx, r.rpcTimeout)
+	ctx, cancel := multinode.MakeQueryCtx(callerCtx, r.GetChStopInflight(), r.rpcTimeout)
 	defer cancel()
 
 	if r.ws == nil && r.http == nil {
```

Comment:
Seems like there is an old race condition. We are using `acquireQueryCtx` to safely get copy of WS and HTTP clients, but on Dial we are modifying them without lock. 
Also we might want to use atomic.Pointer to safely read/write to r.ws and r.http.

Reply:
Changed to use atomic.Pointer!